sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -1520,6 +1521,292 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
             }
         }
         #endregion GetBlockList
+
+        #region SetExpiry
```

Comment:
Note that Set Expiry only works for Block Blobs.  I'm not sure if it will work with other blob types in the future, but if it does, we can move these APIs to BlobBaseClient.